app/Services/Account/Activity/Activity/CreateActivity.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Services\Account\Activity\Activity;
+
+use App\Services\BaseService;
+use App\Models\Account\Activity;
+use App\Models\Account\ActivityType;
+use App\Models\Journal\JournalEntry;
+use App\Models\Instance\Emotion\Emotion;
+
+class CreateActivity extends BaseService
+{
+    /**
+     * Get the validation rules that apply to the service.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'activity_type_id' => 'nullable|integer',
+            'summary' => 'required|string:255',
+            'description' => 'nullable|string:400000000',
+            'date' => 'required|date_format:Y-m-d',
+            'emotions' => 'nullable|array',
+        ];
+    }
+
+    /**
+     * Create an activity.
+     *
+     * @param array $data
+     * @return Activity
+     */
+    public function execute(array $data) : Activity
+    {
+        $this->validate($data);
+
+        if ($data['activity_type_id']) {
+            ActivityType::where('account_id', $data['account_id'])
+                ->findOrFail($data['activity_type_id']);
+        }
+
+        $activity = Activity::create([
+            'account_id' => $data['account_id'],
+            'activity_type_id' => $this->nullOrValue($data, 'activity_type_id'),
+            'summary' => $data['summary'],
+            'description' => $this->nullOrValue($data, 'description'),
+            'happened_at' => $data['date'],
+        ]);
+
+        if (! empty($data['emotions'])) {
+            if ($data['emotions'] != '') {
+                $this->addEmotions($data['emotions'], $activity);
+            }
+        }
+
+        // Log a journal entry
+        (new JournalEntry)->add($activity);
+
+        return Activity::find($activity->id);
```

Comment:
Why not just return `$activity`? ( maybe after a refresh())

Reply:
@asbiin Because `create` returns an object with only the fields that have been modified, whereas this line returns the object with all the attributes. I don't know what `refresh()` does.

Reply:
From [Lararavel documentation](https://laravel.com/docs/5.8/eloquent#retrieving-models):
```
The refresh method will re-hydrate the existing model using fresh data from the database. In addition, all of its loaded relationships will be refreshed as well:
```
So it will do almost the same, but `$activity->refresh();` is more elegant, I think.